requests_oauthlib/oauth2_session.py

```diff
@@ -1,15 +1,33 @@
 from __future__ import unicode_literals
 
 import logging
+import warnings
 
 from oauthlib.common import generate_token, urldecode
 from oauthlib.oauth2 import WebApplicationClient, InsecureTransportError
 from oauthlib.oauth2 import TokenExpiredError, is_secure_transport
+
 import requests
 
 log = logging.getLogger(__name__)
 
 
+# Preserve so we can call it.
+old_showwarning = warnings.showwarning
+
+def logwarning(message, category, filename, lineno, file=None):
+    """
+    Log warnings as well as print them.
+    """
+    log.warning(
+        '%s:%s: %s:%s' % 
+        (filename, lineno, category.__name__, message))
+    old_showwarning(message, category, filename, lineno, file=file)
+
+# Install our new handler.
+warnings.showwarning = logwarning
```

Comment:
Sure, but it is still unnecessary for us to do this. We can just use a regular function, rather than change global state.